### PR TITLE
Create a log file of all files installed.

### PR DESCRIPTION
**Pull request by jpakkane**

### CODE REVIEW

File: mesonbuild/scripts/meson_install.py [hunk 1]
```diff
@@ -224,11 +234,19 @@ def install_targets(d):
...
 
 def run(args):
+    global install_log_file
     if len(args) != 1:
         print('Installer script for Meson. Do not run on your own, mmm\'kay?')
         print('meson_install.py [install info file]')
     datafilename = args[0]
+    private_dir = os.path.split(datafilename)[0]
+    log_dir = os.path.join(private_dir, '../meson-logs')
+    install_log_file = open(os.path.join(log_dir, 'install-log.txt'), 'w')
```
Comment (foolswood) [inline mesonbuild/scripts/meson_install.py]: Context manager for file?

File: mesonbuild/scripts/meson_install.py [hunk 2]
```diff
@@ -19,6 +19,12 @@
 from mesonbuild.scripts import depfixer
 from mesonbuild.scripts import destdir_join
 
+install_log_file = None
+
+def log_installed_file(fname):
```
Comment (foolswood) [inline mesonbuild/scripts/meson_install.py]: This is append_to_log()?

File: mesonbuild/scripts/meson_install.py [hunk 3]
```diff
@@ -224,11 +234,19 @@ def install_targets(d):
...
 def run(args):
+    global install_log_file
     if len(args) != 1:
         print('Installer script for Meson. Do not run on your own, mmm\'kay?')
         print('meson_install.py [install info file]')
     datafilename = args[0]
+    private_dir = os.path.split(datafilename)[0]
+    log_dir = os.path.join(private_dir, '../meson-logs')
+    install_log_file = open(os.path.join(log_dir, 'install-log.txt'), 'w')
+    install_log_file.write('# List of files installed by Meson.\n')
```
Comment (foolswood) [inline mesonbuild/scripts/meson_install.py]: If the above is just write a line to log, then this can use that?

**Comment by foolswood** [comment]
Is there a particular reason not to use python logging?

**Comment by jpakkane** [comment]
We don't use Python logging because the log format must be exactly this. We plan to use it later to create an uninstall target.

**Comment by foolswood** [comment]
In theory you could set a formatter, but I don't think it'd buy you anything.
I get why you're going straight to file and agree that it seems like the clearest way.

**Comment by jpakkane** [comment]
Updated with fixes.

File: mesonbuild/scripts/meson_install.py [hunk 4]
```diff
@@ -224,11 +236,20 @@ def install_targets(d):
...
     datafilename = args[0]
-    do_install(datafilename)
+    private_dir = os.path.split(datafilename)[0]
+    log_dir = os.path.join(private_dir, '../meson-logs')
+    with open(os.path.join(log_dir, 'install-log.txt'), 'w') as lf:
+        install_log_file = lf
+        append_to_log('# List of files installed by Meson')
+        append_to_log('# Does not contain files installed by custom scripts.')
+        do_install(datafilename)
+        install_log_file.close()
```
Comment (foolswood) [inline mesonbuild/scripts/meson_install.py]: If you're using the context you don't need to do this.

**Comment by foolswood** [comment]
Given how side effecty this file is generally I can understand only testing it at integration.
I'm happy with the diff (for all the value that some random guy on the internet approving your stuff has).

**Pull request merged**